### PR TITLE
Generated Recipes Screen

### PR DESCRIPTION
**Pull request by ohamz**
Implement the UI for displaying the recipes recommended in a new subscreen.

- [x] Implement UI for generated recipes screen.
- [x] Link frontend with backend implementation.
- [x]  Test all.
- [x] Get required coverage.

### CODE REVIEW

**Comment by ohamz** [comment]
I have to wait for the recommendation system backend to be implemented before completing this PR.

**Comment by ohamz** [comment]
Ready for review :)


@fdejean Could you check if the backend works please, because I have a feeling it doesn't (unless I forgot something when linking it to the UI).

**Comment by fdejean** [comment]
> Ready for review :)
> 
> @fdejean Could you check if the backend works please, because I have a feeling it doesn't (unless I forgot something when linking it to the UI).

Yes, there was in issue with database missing an IDs field. 
I fixed it work now!

File: app/src/main/java/com/android/feedme/model/viewmodel/GenerateViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+   * A function that fetches the generated recipes
+   *
+   * @param ingredientIds: the unique IDs of the recipes we want to fetch
+   * @param profile: the profile of the user
+   */
+  fun fetchGeneratedRecipes(ingredientIds: List<String>, profile: Profile) {
+    _ingredientsIds = ingredientIds
+    _profile = profile
+
+    if (_isStrict.value) {
```
Comment (fdejean) [inline app/src/main/java/com/android/feedme/model/viewmodel/GenerateViewModel.kt:39]: The logic is reversed, by default, the toggle button is on "strict" and it's seem the value is true as when sliding to "extra" is does the strict search.

File: app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt [hunk 2]
```diff
@@ -325,21 +326,31 @@ class RecipeRepository(private val db: FirebaseFirestore) {
...
   ) {
     if (ingredientIds.isEmpty()) {
       onFailure(IllegalArgumentException())
       return
     }
-    db.collection(collectionPath)
-        .whereArrayContainsAny("ingredientIds", ingredientIds)
+    // Create a query to fetch the top rated recipes
+    var queryRef =
+        db.collection(collectionPath).whereArrayContainsAny("ingredientIds", ingredientIds).limit(6)
```
Comment (fdejean) [inline app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt:346]: No, you cannot do that, here you limit the recipes to 6. Hence the ranking will happen only on those 6 recipes and not all the recipe that match criteria. Either increase this limit or do a filtering on the highest ranked recipes first.
The best way will be to utilize SQL firebase or BigQuerry but it was decided that local filtering will work for now

**Review by fdejean — CHANGES_REQUESTED** [review]
Nice work, but some details need to changed for the feature to work as excepted (c.f my comment)

File: app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt (hunk 2 above)
Comment (ohamz) [reply app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt:346]: Hmmm I see, so where do you think can I add this `.limit()` call ? (I did these changes to be able to add the Load more button)

File: app/src/main/java/com/android/feedme/model/viewmodel/GenerateViewModel.kt (hunk 1 above)
Comment (ohamz) [reply app/src/main/java/com/android/feedme/model/viewmodel/GenerateViewModel.kt:39]: Ohhh my bad, nice catch !

File: app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt (hunk 2 above)
Comment (fdejean) [reply app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt:346]: The process is done in "one go" 
You can't really pause it and get a coherent result. 
(Would be the case if it was a not local suggestion)

Hence, there is no load more feature. The output is a complete list, for the UI you could use a LazyList and it should display only the requested element.

Comment (ohamz) [reply app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt:346]: I wanted to implement the Load more button functionality, but I think I will just display all fetched recipes on after the other.

**Review by fdejean — APPROVED** [review]
Nice

**Review by 0marwa — APPROVED** [review]
Good job spotting and fixing the DB bugs, no remarks on my end; the code is well-documented so well done!

**Pull request merged**